mk/artifact.mk

```diff
@@ -0,0 +1,70 @@
+ENABLE_PREBUILT ?= 1
+
+CC ?= gcc
+CROSS_COMPILE ?= riscv-none-elf-
+
+BIN_DIR := $(abspath $(OUT))
+
+TEST_SUITES += \
+	ansibench \
+	rv8-bench
+
+# "ieee754" needs F extension
+# "smolnes", "ticks" have inline assembly and only work in riscv
+TEST_BENCHES += \
+	captcha \
+	donut \
+	fcalc \
+	hamilton \
+	jit \
+	lena \
+	line \
+	maj2random \
+	mandelbrot \
+	nqueens \
+	nyancat \
+	pi \
+	puzzle \
+	qrcode \
+	richards \
+	rvsim \
+	spirograph \
+	uaes
+
+SHELL_HACK := $(shell mkdir -p $(BIN_DIR)/linux-x86-softfp $(BIN_DIR)/riscv32)
+
+ifeq ($(call has, PREBUILT), 1)
+  LATEST_RELEASE := $(shell wget -q https://api.github.com/repos/sysprog21/rv32emu-prebuilt/releases/latest -O- | grep -Po '(?<="tag_name": ").+(?=",)')
```

Comment:
The `-P` option of `grep` is not supported on macOS, the reason is that the BSD version uses the POSIX Compatible Regular Expressions and the GNU version uses the Perl Compatible Regular Expressions (PCRE). Although my BSD version said it is compatible to GNU version ( output of `grep -V`: `grep (BSD grep, GNU compatible) 2.6.0-FreeBSD` ), but it is not support PCRE.

Consider this instead: `wget -q https://api.github.com/repos/sysprog21/rv32emu-prebuilt/releases/latest -O- | grep '"tag_name"' | sed -E 's/.*"tag_name": "([^"]+)".*/\1/'`. This works on both Linux and macOS.

Reply:
Thanks for the information!